python/dgl/contrib/graph_store.py

```diff
@@ -318,7 +318,12 @@ class SharedMemoryStoreServer(object):
     """
     def __init__(self, graph_data, edge_dir, graph_name, multigraph, num_workers, port):
         self.server = None
-        if isinstance(graph_data, (GraphIndex, DGLGraph)):
+        if isinstance(graph_data, GraphIndex):
+            graph_data = graph_data.copyto_shared_mem(edge_dir, _get_graph_path(graph_name))
```

Comment:
We probably should make shared memory as a special device type (similar to kDLCPUPinned in current DLPack). I created a issue in DLPack (https://github.com/dmlc/dlpack/issues/41) for this. This is definitely sth after this release though.

Reply:
yes, I agree.